src/actions/actions.js

```diff
@@ -188,15 +188,30 @@ export function addLanguage(languageDefinition) {
   };
 }
 
+export function updateEditorCursor(
+  editorCursorLine,
+  editorCursorChar,
+  editorCursorForceUpdate = false
+) {
+  return {
+    type: "UPDATE_CURSOR",
+    editorCursorLine,
+    editorCursorChar,
+    editorCursorForceUpdate
+  };
+}
+
 export function evaluateText() {
   return (dispatch, getState) => {
     const { jsmdChunks, kernelState } = getState();
     if (kernelState !== "KERNEL_BUSY") dispatch(setKernelState("KERNEL_BUSY"));
     const cm = window.ACTIVE_CODEMIRROR;
     const doc = cm.getDoc();
     if (!doc.somethingSelected()) {
-      const { line } = doc.getCursor();
-      const activeChunk = getChunkContainingLine(jsmdChunks, line);
+      const activeChunk = getChunkContainingLine(
+        jsmdChunks,
+        getState().editorCursorLine
```

Comment:
we are already calling `getState` / destructuring that above - can we use that same destructuring for `editorCursorLine`?